org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java

```diff
@@ -278,6 +288,44 @@ private static boolean getExtractMethodProposal(IInvocationContext context, ASTN
 		return true;
 	}
 
+	private static boolean getConvertAnonymousClassCreationsToLambdaProposals(IInvocationContext context, ASTNode covering, Collection<CUCorrectionProposal> resultingCollections) {
+		while (covering instanceof Name || covering instanceof Type || covering instanceof Dimension || covering.getParent() instanceof MethodDeclaration

```

Comment:
potential NPE when calling covering.getParent()

Reply:
The covering parameter is checked from the getAssists function before passing [HERE](https://github.com/SummerSun/eclipse.jdt.ls/blob/7aa3df4548b5e2d6ee12a88397ef7d6c61cabbc3/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java#L135) , and the function getConvertAnonymousClassCreationsToLambdaProposals is private used in the same scope. Is that ok?